### PR TITLE
Add `rocksdb` backend for `bee-autopeering` stores

### PR DESCRIPTION
**Pull request by pvdrz**

### CODE REVIEW

File: bee-network/bee-autopeering/Cargo.toml [hunk 1]
```diff
@@ -26,10 +26,11 @@ num-derive = { version = "0.3.3", default-features = false  }
 num-traits = { version = "0.2.14" , default-features = false }
 priority-queue = { version = "1.2.0" , default-features = false }
 prost = { version = "0.8", default-features = false, features = [ "std" ] }
+rocksdb = { version = "0.17.0", default-features = false, optional = true }
```
Comment (thibault-martinez) [inline bee-network/bee-autopeering/Cargo.toml]: Order!

File: bee-network/bee-autopeering/src/peer/mod.rs [hunk 2]
```diff
@@ -145,6 +145,14 @@ impl Peer {
     pub(crate) fn into_id(self) -> PeerId {
         self.peer_id
     }
+
+    pub(crate) fn as_bytes(&self) -> Vec<u8> {
```
Comment (thibault-martinez) [inline bee-network/bee-autopeering/src/peer/mod.rs]: `to_bytes` maybe ? `as_` usually refers to costless interpretation

File: bee-network/bee-autopeering/src/peer/stores/rocksdb.rs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        let db = DBWithThreadMode::open_cf(&config.options, &config.path, &[ACTIVE_PEERS_CF, REPLACEMENTS_CF])
+            .expect("error opening peer store");
+
+        Self {
+            db: Arc::new(db),
+            config,
+        }
+    }
+
+    fn store_active(&self, active_peer: ActivePeer) {
```
Comment (thibault-martinez) [inline bee-network/bee-autopeering/src/peer/stores/rocksdb.rs]: I know the trait was already like that but I don't think we should panic all backend errors

File: bee-network/bee-autopeering/src/peer/stores/sled.rs [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+impl PeerStore for SledPeerStore {
+    type Config = SledPeerStoreConfig;
+
+    type Error = sled::Error;
+
+    fn new(config: Self::Config) -> Result<Self, Self::Error> {
+        let db = config.open()?;
+
+        db.open_tree("active_peers")?;
```
Comment (thibault-martinez) [inline bee-network/bee-autopeering/src/peer/stores/sled.rs]: Use the constants

File: bee-network/bee-autopeering/src/peer/stores/rocksdb.rs [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    fn delete(&self, _: &PeerId) -> Result<bool, Self::Error> {
+        unimplemented!("no need for single entry removal at the moment")
+    }
+
+    fn delete_all(&self) -> Result<(), Self::Error> {
+        self.db.drop_cf(ACTIVE_PEERS_CF)?;
+        self.db.create_cf(ACTIVE_PEERS_CF, &self.config.options)?;
+
+        self.db.drop_cf(REPLACEMENTS_CF)?;
+        self.db.create_cf(REPLACEMENTS_CF, &self.config.options)?;
```
Comment (thibault-martinez) [inline bee-network/bee-autopeering/src/peer/stores/rocksdb.rs]: I feel like I remember there was an issue proceeding that way, and we had to come up with `truncate` that uses `delete_range_cf`

**Review by grtlr — DISMISSED** [review]

File: bee-network/bee-autopeering/Cargo.toml [hunk 6]
```diff
@@ -46,3 +47,6 @@ prost-build = { version = "0.8", default-features = false }
 [[example]]
 name = "node"
 
+[features]
+default = [ "rocksdb" ]
+in-memory = [ ]
```
Comment (Alex6323) [inline bee-network/bee-autopeering/Cargo.toml:52]: Shouldn't this section also explicitly announce the `sled` and `rocksdb` feature, and then make `default` point to the `rocksdb` feature? I think that would be nicer from a UX pov.

Comment (pvdrz) [reply bee-network/bee-autopeering/Cargo.toml:52]: sadly this is illegal for now: https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies

File: bee-network/bee-autopeering/src/peer/stores/sled.rs (hunk 4 above)
Comment (pvdrz) [reply bee-network/bee-autopeering/src/peer/stores/sled.rs]: @Alex6323 use the constants :trollface:

File: bee-network/bee-autopeering/src/peer/stores/rocksdb.rs [hunk 7]
```diff
@@ -134,12 +143,89 @@ impl PeerStore for RocksDbPeerStore {
...
+    fn run_with_peer_store_in_path<P: AsRef<Path> + Copy>(path: P, f: fn(RocksDbPeerStore)) {
+        let mut options = Options::default();
+        options.create_if_missing(true);
+        options.create_missing_column_families(true);
+
+        let config = RocksDbPeerStoreConfig::new(path, options);
+        let peer_store = RocksDbPeerStore::new(config).unwrap();
+        let janitor = Janitor(path);
+        f(peer_store);
+        drop(janitor);
```
Comment (grtlr) [inline bee-network/bee-autopeering/src/peer/stores/rocksdb.rs:181]: Isn't the drop implicit at the end of the scope?

File: bee-network/bee-autopeering/src/peer/stores/rocksdb.rs [hunk 8]
```diff
@@ -134,12 +143,89 @@ impl PeerStore for RocksDbPeerStore {
...
+        let cf = self.open_cf(REPLACEMENTS_CF);
+        self.db.delete_range_cf(&cf, [0; 32], [0xff; 32])?;
 
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    struct Janitor<P: AsRef<Path>>(P);
```
Comment (grtlr) [inline bee-network/bee-autopeering/src/peer/stores/rocksdb.rs:158]: I like that ❤️

**Review by grtlr — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**